### DISCUSSION TITLE
Why constructors in Ć _never_ take arguments ?

### DISCUSSION THREAD

**Discussion by andr1972** [pfusik/cito#58] [Q&A]
In documentation is
Constructors in Ć _never_ take arguments. This promotes reuse of existing
objects instead of creating a bunch of single-use objects.
If you need to initialize the object with some outside data,
create a method such as `Init`.

Could you elaborate on this thought , why is better, why this promotes reuse of existing
objects on example?

**Comment by pfusik** [comment]
Actually there are two reasons for no constructor parameters:

1. Ć doesn't support overloading, because C and JavaScript don't support it.

2. Java is often considered slow, because of poorly-performing code that does too many allocations. Example:

```java
for (Foo f : largeCollection) {
    Bar b = new Bar(f.getQuux(), f.getFred());
    ...
}
```

instead of:

```java
Bar bar = new Bar();
for (Foo f : largeCollection) {
    b.setQuux(f.getQuux());
    b.setFred(f.getFred());
    ...
}
```

This is what I meant by reusing existing objects.

Having said that, Ć will probably support C#-style field initializers:

```csharp
Point() p = { X = 3, Y = 6 };
```

These are not constructor arguments. These are values assigned to fields after the constructor completes. Basically, it's a syntax sugar for:

```csharp
Point() p;
p.X = 3;
p.Y = 6;
```

It has an advantage that you can initialize any field without writing code for it in the constructor.

**Comment by pfusik** [comment]
Since 2.1.1, Fusion supports object literals:
```csharp
Point() p = { X = 3, Y = 6 };
```

and dynamic object literals:
```csharp
Point# p = new Point { X = 3, Y = 6 };
```
